src/mail-app/search/view/SearchViewModel.ts

```diff
@@ -524,10 +524,17 @@ export class SearchViewModel {
 			this.freeToAskAboutExtendingIndex = true
 			if (confirmed) {
 				this._startDate = startDate
-				this.indexerFacade.extendMailIndex(startDate.getTime()).then(() => {
-					this.updateSearchUrl()
-					this.updateUi()
+
+				const searchRestriction = this.getRestriction()
+				this.indexerFacade.extendMailIndex(startDate.getTime()).then(async () => {
+					// don't do anything further if the search parameters were changed and thus createRestriction was called again
+					if (searchRestriction !== this.getRestriction()) {
```

Comment:
this is a referential comparison but this is intentional? memoization is done on the whole URL but the restriction is only computed from part of it so it might sometimes be not referentially equal